src/store/wizardSlice.ts

```diff
@@ -87,6 +88,14 @@ export type wizardState = {
     useLatest: boolean;
     snapshotDate: string;
   };
+  users: User[];
+  userInfo: {
+    userName: string;
+    password: string;
+    confirmPassword: string;
+    sshKey: string;
+    administrator: boolean;
+  };
```

Comment:
Already asked a bit offline - if I understand correctly the UI will support just one user for now and I understand that the API type of User doesn't include all the data we want to keep in the store.

I think we should either have a single user and later expand it to an array, or we could make the structure a bit more future proof and create an array right away:
```
users: UserWithAdditionalInfo[];
```

With the type looking something like:
```
export type UserWithAdditionalInfo = {
  name: User['name'];
  ssh_key: User['ssh_key'];
  randomOtherInfo: string;
  ...
```

What do you think?

Reply:
I added the userState structure to make it easy to fill every field and validate it 
```
  userState: {
    users: [],
    userInfo: {
      index: 0,
      name: '',
      password: '',
      confirmPassword: '',
      ssh_key: '',
      administrator: false,
    },
  },
```